### PR TITLE
Core/Spell: Rewrite Steady Shot damage

### PR DESCRIPTION
**Pull request by robinsch**
**Changes proposed**:
- Steady Shot now deals the correct damage when using Heirloom Weapons.

**Target branch(es)**: 335

**Tests performed**: (Does it build, tested in-game, etc)
Tested in game.

**Known issues and TODO list**:
None

### CODE REVIEW

File: src/server/game/Spells/SpellEffects.cpp [hunk 1]
```diff
@@ -596,17 +596,13 @@ void Spell::EffectSchoolDMG(SpellEffIndex effIndex)
...
-                                damage += irand(int32(dmg_min), int32(dmg_max));
-                            damage += int32(caster->GetAmmoDPS() * weaponTemplate->Delay * 0.001f);
-                        }
+                        float dmg_min = caster->GetWeaponDamageRange(RANGED_ATTACK, MINDAMAGE);
+                        float dmg_max = caster->GetWeaponDamageRange(RANGED_ATTACK, MAXDAMAGE);
+                        if (dmg_max == 0.0f && dmg_min > dmg_max)
+                            damage += int32(dmg_min);
+                        else
+                            damage += irand(int32(dmg_min), int32(dmg_max));
+                        damage += int32(caster->GetAmmoDPS() * caster->GetAttackTime(RANGED_ATTACK) * 0.001f);
```
Comment (Shauren) [inline src/server/game/Spells/SpellEffects.cpp]: GetAttackTime includes haste

Comment (robinsch) [inline src/server/game/Spells/SpellEffects.cpp]: No, it returns base attack time, same as item->Delay

Comment (Shauren) [inline src/server/game/Spells/SpellEffects.cpp]: hmm ok

**Comment by Demonid** [comment]
This gotta be the fastesr PR merged xD

Gj both

**Pull request merged**